src/BloomExe/ImageProcessing/ImageServer.cs

```diff
@@ -76,41 +76,42 @@ protected override bool ProcessRequest(IRequestInfo info)
 			if (!_useCache)
 				return false;
 
-			var r = GetLocalPathWithoutQuery(info);
+			var imageFile = GetLocalPathWithoutQuery(info);
 
 			// only process images
-			var isSvg = r.EndsWith(".svg", StringComparison.OrdinalIgnoreCase);
-			if (!IsImageTypeThatCanBeDegraded(r) && !isSvg)
+			var isSvg = imageFile.EndsWith(".svg", StringComparison.OrdinalIgnoreCase);
+			if (!IsImageTypeThatCanBeDegraded(imageFile) && !isSvg)
 				return false;
 
-			r = r.Replace("thumbnail", "");
+			imageFile = imageFile.Replace("thumbnail", "");
+
+			var processImage = !isSvg;
 
 			// This happens with the new way we are serving css files
-			if (!File.Exists(r))
+			if (!File.Exists(imageFile))
 			{
-				var fileName = Path.GetFileName(r);
+				var fileName = Path.GetFileName(imageFile);
 				var sourceDir = FileLocator.GetDirectoryDistributedWithApplication("BloomBrowserUI");
-				r = Directory.EnumerateFiles(sourceDir, fileName, SearchOption.AllDirectories).FirstOrDefault();
+				imageFile = Directory.EnumerateFiles(sourceDir, fileName, SearchOption.AllDirectories).FirstOrDefault();
+
+				// image file not found
+				if (string.IsNullOrEmpty(imageFile)) return false;
+
+				// BL-2368: do not process files from the BloomBrowserUI directory
```

Comment:
Maybe say briefly why, e.g....(a) we assume the images we ship as part of the program are already in the best possible state (b) making parts of these images transparent may cause problems like BL-2368